nifi-nar-bundles/nifi-graph-bundle/nifi-graph-processors/src/main/java/org/apache/nifi/processors/graph/ExecuteGraphQueryRecord.java

```diff
@@ -0,0 +1,265 @@
+
+    public static final PropertyDescriptor CLIENT_SERVICE = new PropertyDescriptor.Builder()
+            .name("client-service")
+            .displayName("Client Service")
+            .identifiesControllerService(GraphClientService.class)
+            .addValidator(Validator.VALID)
+            .required(true)
+            .build();
+
+    public static final PropertyDescriptor READER_SERVICE = new PropertyDescriptor.Builder()
+            .name("reader-service")
+            .displayName("Record Reader")
+            .description("The record reader to use with this processor.")
+            .identifiesControllerService(RecordReaderFactory.class)
+            .required(true)
+            .addValidator(Validator.VALID)
+            .build();
+
+    public static final PropertyDescriptor WRITER_SERVICE = new PropertyDescriptor.Builder()
+            .name("writer-service")
+            .displayName("Failed Record Writer")
+            .description("The record writer to use for writing failed records.")
+            .identifiesControllerService(RecordSetWriterFactory.class)
+            .required(true)
+            .addValidator(Validator.VALID)
+            .build();
+
+    private static String loadScript(String path) {
+        try {
+            return IOUtils.toString(ExecuteGraphQueryRecord.class.getResourceAsStream(path), StandardCharsets.UTF_8);
+        } catch (Exception ex) {
+            throw new ProcessException(ex);
+        }
+    }
+
+    public static final PropertyDescriptor SUBMISSION_SCRIPT = new PropertyDescriptor.Builder()
+            .name("record-script")
+            .displayName("Graph Record Script")
+            .description("Script to perform the business logic on graph, using flow file attributes and custom properties " +
+                    "as variable-value pairs in its logic.")
+            .required(true)
+            .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+            .addValidator(StandardValidators.NON_EMPTY_EL_VALIDATOR)
+            .build();
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName)
+                .required(false)
+                .addValidator(StandardValidators.ATTRIBUTE_KEY_PROPERTY_NAME_VALIDATOR)
+                .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+                .dynamic(true).addValidator(StandardValidators.NON_EMPTY_VALIDATOR).build();
+    }
+
+    public static final List<PropertyDescriptor> DESCRIPTORS = Collections.unmodifiableList(Arrays.asList(
+            CLIENT_SERVICE, READER_SERVICE, WRITER_SERVICE, SUBMISSION_SCRIPT
+    ));
+
+    public static final Relationship REL_SUCCESS = new Relationship.Builder().name("original").description("The original flowfile").build();
```

Comment:
Make these builders use multiple indented lines for legibility.

Reply:
Multiline builders added